src/components/LmsConfigurations/DegreedIntegrationConfigForm.test.jsx

```diff
@@ -0,0 +1,67 @@
+import {
+  render, fireEvent, screen, act,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+import DegreedIntegrationConfigForm from './DegreedIntegrationConfigForm';
+import LmsApiService from '../../data/services/LmsApiService';
+
+jest.mock('../../data/services/LmsApiService', () => ({
+  postNewDegreedConfig: jest.fn(),
+  updateDegreedConfig: jest.fn(),
+}));
+
+const enterpriseId = 'test-enterprise';
+
+describe('<DegreedIntegrationConfigForm />', () => {
+  test('renders Degreed Config Form', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    // Verify all expected fields are present.
+    screen.getByLabelText('Active');
+    screen.getByLabelText('Degreed User ID');
+    screen.getByLabelText('Degreed User Password');
+    screen.getByLabelText('Degreed Instance URL');
+    screen.getByLabelText('Degreed Organization Code');
+    screen.getByLabelText('API Client ID');
+    screen.getByLabelText('API Client Secret');
+  });
+
+  test('required fields show as invalid when not filled in', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    fireEvent.click(screen.getByText('Submit'));
+
+    expect(screen.getByLabelText('Degreed User ID')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed User Password')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed Instance URL')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('Degreed Organization Code')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('API Client ID')).toHaveClass('is-invalid');
+    expect(screen.getByLabelText('API Client Secret')).toHaveClass('is-invalid');
+  });
+
+  test('submit calls postNewDegreedConfig when config is not present', () => {
+    render(<DegreedIntegrationConfigForm enterpriseId={enterpriseId} />);
+    act(() => {
+      fireEvent.change(screen.getByLabelText('Degreed User ID'), {
+        target: { value: 'testing' },
```

Comment:
I think it's worth changing these values to be different for each one, otherwise you don't know if the form has the correct value or if it mistakenly has a value that's supposed to be in a different field.